### ISSUE TITLE
editor type: 'handsontable' cell value while editing shows [object Object]

### ISSUE THREAD

**Issue by akoskm** [handsontable/handsontable#7449]
I have a combination of two tables:
1. **Main** table
2. **Source** table

**Main** table has a text and a `handsontable` type column
![image](https://user-images.githubusercontent.com/3111627/103130462-1620f300-469d-11eb-8c9f-b413ac3d7ec2.png)

where the possible values for the `handsontable` column are coming from the **Source** table:

![image](https://user-images.githubusercontent.com/3111627/103130554-6ef08b80-469d-11eb-8939-afd310f7b450.png)

Here's how the definition looks like for the **Main** table:

```
    { type: 'text' },
    {
      type: 'handsontable',
      handsontable: {
        colHeaders:
          headerLevelMappings['fig_1'],
        data: tableState['fig_1'].levels,
        getValue,
        colWidths: [120]
      },
      renderer: arrayToStringRenderer // see arrayToStringRenderer below
    }


function arrayToStringRenderer(
  _instance: Handsontable,
  TD: HTMLTableCellElement,
  _row: number,
  _col: number,
  _prop: string | number,
  value: Handsontable.CellValue
) {
  const tdValue = value === null ? '' : Object.values(value).join(',');
  TD.innerHTML = tdValue;
}
```

`arrayToStringRenderer` helps to turn the array selected with the handsontable editor into a string and it's displayed correctly (you can see that on the first screenshot).

The problem appears when I try to edit a handsontable column that has a value set. When I enter edit mode I see the following:

![image](https://user-images.githubusercontent.com/3111627/103130653-e9b9a680-469d-11eb-8c59-cc90bbf36001.png)

This leads to another problem when I try to update any of the values in the handsontable column after selecting a new value from the handsontable editor the value in the Main table doesn't update.

I think the source of the problem is having `[object Object]` as a value while editing. When I select a different row, its representation is also `[object Object]` so the editor doesn't notice the change.

**Comment by AMBudnik** [comment]
Hi @akoskm 

At the first glance, I can see that you've made a custom renderer but the value for an editor is still kept as an object, while it can be only a string, number, or null by default. 

If you would like to allow changing the value I recommend [creating a custom editor](https://handsontable.com/docs/8.2.0/tutorial-cell-editor.html). You may find it more useful to extend the [HandsontableEditor](https://github.com/handsontable/handsontable/blob/b7618fc4213d3b2ab32ba62ff6f1d57cd1af6c49/src/editors/handsontableEditor.js) instead of the [TextEditor](https://github.com/handsontable/handsontable/blob/b7618fc4213d3b2ab32ba62ff6f1d57cd1af6c49/src/editors/textEditor.js).

**Comment by akoskm** [comment]
@AMBudnik thanks for your response. I created a custom editor but I'm not exactly sure at which point I can alter what's displayed in the column while editing and what's the original column value. Here's my editor, see my comments inline.

```
import Handsontable from 'handsontable';

export default class CustomHandsontableEditor extends Handsontable.editors
  .HandsontableEditor {

  // this is the same function from the code I shared in the first snippet
  getValue(this: Handsontable) {
    // *** copy-paste from your docs BEGINS here ***
    // https://handsontable.com/docs/8.2.0/demo-handsontable.html
    const selection = this.htEditor.getSelectedLast();
    if (!selection) return '';

    // coordinates (negative values)
    const backstageRow = this.htEditor.getSourceDataAtRow(
      Math.max(selection[0], 0)
    ) as number[];
    // *** copy-paste from your docs ENDS here ***
    // after this line, "backstageRow" contains the array I want to save as the value for this cell

    // I don't need "test 1", "test 2" in the editor, see my first two screenshots
    const value = backstageRow.slice(1);

    // I want the cell to display the stringified version of the "value" array
    this.TD.value = Object.values(value).join(',');
    // but I want retain the array as the original value for this column, so I have to return it
    return value;
  }
}
```

**Comment by AMBudnik** [comment]
Hi @akoskm 

I've read it all again and I'm wondering if that https://jsfiddle.net/handsoncode/y1gdbq72/ would work for you. Let me know if I'm missing anything here.

**Comment by akoskm** [comment]
Thanks for this great example @AMBudnik. 🙇 

The problem was in my code:
In the main table, I had an `afterChange` event that used `reduce` on the selected array (this was coming from the Source table) and turn it into an _**object**_. I guess this was the cause for the `[object Object]` I saw in the table when I double-clicked the cell.

I did this because finally when I use the selection I need it as an object.

I changed my approach and I'm simply not transforming the data after every `afterChange`, but right before I need the selection as an object.

Thank you for your help.
